### ISSUE TITLE
Split API documentation into "safe" and "core"

### ISSUE THREAD

**Issue by astrojuanlu** [poliastro/poliastro#1183]
- Split API reference in two, like it was before: https://docs.poliastro.space/en/v0.14.0/api/safe/safe_index.html and https://docs.poliastro.space/en/v0.14.0/api/core/core_index.html
- Put redirects in place: some pages changed their name, and I would like to avoid broken links. Read the Docs allows this easily 🤓

**Comment by astrojuanlu** [comment]
- I forgot to actually rename "Getting started" to "Installation"

**Comment by astrojuanlu** [comment]
- Splitting the API reference in two looks rather difficult with sphinx-autoapi, or at least I couldn't figure out how.
- I will put the redirects after the beta is in place.
- I did the renaming in #1206.

**Comment by astrojuanlu** [comment]
I investigated this a bit. Still getting familiar with the internals of sphinx-autoapi, did some minor improvements in #1432. I think we might need to write parts of the API docs manually, not a big deal.

**Comment by astrojuanlu** [comment]
After merging #1446, the problem of auto API docs became more pressing.

**Comment by astrojuanlu** [comment]
I spent a fair bit of time getting familiar with sphinx-autoapi. I found some issues along the way https://github.com/readthedocs/sphinx-autoapi/issues/325, re-discovered a few old ones (https://github.com/readthedocs/sphinx-autoapi/issues/317, https://github.com/readthedocs/sphinx-autoapi/issues/285), and I am still figuring out why the generated sources do not contain the `Orbit` documentation at the right place.

It is possible to tweak the structure manually by replacing the generated `toctree` here by more fine-grained headers and groups:

https://github.com/poliastro/poliastro/blob/1cc6b04efef353d52ed2f15991e58b3076b399a4/docs/source/_autoapi_templates/index.rst?plain=1#L26-L33

Hopefully, that will be enough. However, the more I look at the docs, the less happy I am with dumping every single module there. I will try to see how far I can go without having to keep the generated sources in version control.
